wandb/sdk/internal/sender.py

```diff
@@ -890,16 +950,117 @@ def send_stats(self, record: "Record") -> None:
         self._fs.push(filenames.EVENTS_FNAME, json.dumps(row))
         # TODO(jhr): check fs.push results?
 
+    def _output_raw_finish(self) -> None:
+        for stream, output_raw in self._output_raw_streams.items():
+            output_raw._stopped.set()
+
+            # shut down threads
+            output_raw._writer_thr.join(timeout=5)
+            if output_raw._writer_thr.is_alive():
+                logger.info("processing output...")
+                output_raw._writer_thr.join()
```

Comment:
wonder why we don't hit this one

Reply:
because it would mean that it took more than 5 seconds to flush the writer thread.  which would mean we loaded it up with lots of work.  This is possible to do but out of scope for the current tests .

Reply:
oh ok... thanks... no i think it's fine not to hit it if it takes so much time